sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallMediaAsync.java

```diff
@@ -125,55 +123,55 @@ public Mono<Void> recognize(RecognizeOptions recognizeOptions) {
      * @param recognizeOptions Different attributes for recognize.
      * @return Response for successful recognize request.
      */
-    public Mono<Response<Void>> recognizeWithResponse(RecognizeOptions recognizeOptions) {
+    public Mono<Response<Void>> recognizeWithResponse(CallMediaRecognizeOptions recognizeOptions) {
         return withContext(context -> recognizeWithResponseInternal(recognizeOptions, context));
     }
 
-    Mono<Response<Void>> recognizeWithResponseInternal(RecognizeOptions recognizeOptions, Context context) {
+    Mono<Response<Void>> recognizeWithResponseInternal(CallMediaRecognizeOptions recognizeOptions, Context context) {
         try {
             context = context == null ? Context.NONE : context;
 
-            RecognizeConfigurations recognizeConfigurations = recognizeOptions.getRecognizeConfiguration();
-            DtmfConfigurationsInternal dtmfConfigurationsInternal = null;
-            if (recognizeConfigurations.getDtmfConfigurations() != null) {
-                DtmfConfigurations dtmfConfigurations = recognizeConfigurations.getDtmfConfigurations();
-                dtmfConfigurationsInternal = new DtmfConfigurationsInternal()
-                    .setMaxTonesToCollect(dtmfConfigurations.getMaxTonesToCollect());
+            if (recognizeOptions instanceof CallMediaRecognizeDtmfOptions) {
 
-                if (dtmfConfigurations.getInterToneTimeoutInSeconds() != null) {
-                    dtmfConfigurationsInternal.setInterToneTimeoutInSeconds((int) dtmfConfigurations.getInterToneTimeoutInSeconds().getSeconds());
+                DtmfOptionsInternal dtmfOptionsInternal = new DtmfOptionsInternal();
+
+                if (((CallMediaRecognizeDtmfOptions) recognizeOptions).getInterToneTimeout() != null) {
+                    dtmfOptionsInternal.setInterToneTimeoutInSeconds((int) ((CallMediaRecognizeDtmfOptions) recognizeOptions).getInterToneTimeout().getSeconds());
```

Comment:
we are finding an issue where this gets defaulted to 0, we are syncing on what the behavior should be. I'll keep you posted.